### PR TITLE
[FIX] dashboard widget sorting ux (Podio bug_57)

### PR DESCRIPTION
**Pull request by yosokus**

### CODE REVIEW

File: Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ * License along with the script.                                         *
+ * If not, see http://www.gnu.org/licenses/lgpl.html                      *
+ *                                                                        *
+ * The TYPO3 project - inspiring people to share!                         *
+ *                                                                        */
+
+class GridAttributesViewHelper extends \TYPO3\CMS\Fluid\ViewHelpers\Be\AbstractBackendViewHelper
+{
+    /**
+     * Returns grid item attributes
```
Comment (MattiasNilsson) [inline Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php]: Add some more documentation about what actually take places in the code.

File: Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    public function render($widgetSetting, $index, $className = 'grid-item')
+    {
+        $numberOfCols = 3;
+        $widgetIdentifier = $widgetSetting->getWidgetIdentifier();
+        $widget = $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['dashboard']['widgets'][$widgetIdentifier];
+        list($width, $height) = explode('x', $widget['size']);
+        $width = $this->getItemWidth((int)$width, $numberOfCols);
+        $height = $this->getItemHeight((int)$height);
+
+        $attributes = 'class="' . $className . ' col-md-' . ($width * 4) . ' height-' . $height . '"' .
```
Comment (MattiasNilsson) [inline Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php]: Use sprintf for easier reading here.

File: Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            ' data-id="' . $widgetIdentifier . '-' . $widgetSetting->getUid() . '"'.
+            ' data-w="' . $width . '" data-h="' . $height. '"';
```
Comment (MattiasNilsson) [inline Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php]: data-h => data-height
data-w => data-width

File: Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            $row = 0;
+        } else {
+            $col = $index % $numberOfCols;
+            $row = intval($index / $numberOfCols);
+        }
+        $attributes .= ' data-x="' . $col . '" data-y="' . $row . '"';
+        return $attributes;
+    }
+
+    /**
```
Comment (MattiasNilsson) [inline Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php:70]: Add comment

File: Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+     * @param int $width
```
Comment (MattiasNilsson) [inline Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php]: integer

File: Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+     * @param int $numberOfCols
```
Comment (MattiasNilsson) [inline Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php]: integer

File: Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+            $validWith = $numberOfCols;
+        } elseif ($width) {
+            $validWith = $width;
+        } else {
+            $validWith = 1;
+        }
+        return $validWith;
+    }
+
+    /**
```
Comment (MattiasNilsson) [inline Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php:89]: Add comment

File: Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+     * @param int $height
```
Comment (MattiasNilsson) [inline Classes/ViewHelpers/Be/DashboardWidget/GridAttributesViewHelper.php]: integer

File: Resources/Public/JavaScript/GridList.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+define(['jquery', 'draggableGridList', 'jquery-ui/draggable'], function ($) {
```
Comment (MattiasNilsson) [inline Resources/Public/JavaScript/GridList.js:4]: Add documentation about the file

File: Resources/Public/JavaScript/GridList.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+		console.log('GridList initialize START');
+		$(".gridster ul").gridList({
+			direction: 'vertical',
+			lanes: 3,
+			cellHeight: 250
+		},
+		{
+			handle: '.panel-heading'
+		});
+		console.log('GridList initialize END');
```
Comment (MattiasNilsson) [inline Resources/Public/JavaScript/GridList.js]: Remove console.log

File: Resources/Public/JavaScript/jquery.draggableGridList.js [hunk 11]
```diff
@@ -0,0 +1,370 @@
+// It does not try to register in a CommonJS environment since jQuery is not
```
Comment (MattiasNilsson) [inline Resources/Public/JavaScript/jquery.draggableGridList.js]: Use proper docs:
/**
*/

File: Resources/Public/JavaScript/jquery.draggableGridList.js [hunk 12]
```diff
@@ -0,0 +1,370 @@
...
+			}
+			this._createGridSnapshot();
+			this.gridList.resizeGrid(this.options.lanes);
+			this._updateGridSnapshot();
+
+			this.reflow();
+		},
+
+		resizeItem: function(element, size) {
+			/**
```
Comment (MattiasNilsson) [inline Resources/Public/JavaScript/jquery.draggableGridList.js]: This should go before the function

File: Resources/Public/JavaScript/jquery.draggableGridList.js [hunk 13]
```diff
@@ -0,0 +1,370 @@
...
+			this.render();
+		},
+
+		render: function() {
+			this._applySizeToItems();
+			this._applyPositionToItems();
+		},
+
+		_bindMethod: function(fn) {
+			/**
```
Comment (MattiasNilsson) [inline Resources/Public/JavaScript/jquery.draggableGridList.js]: This should go before the method.

**Comment by MattiasNilsson** [comment]
This looks good! There is just some minor things to tend to. 💯

**Review by MattiasNilsson — CHANGES_REQUESTED** [review]

**Comment by yosokus** [comment]
@MattiasNilsson. Please review and provide feedback.

**Comment by MattiasNilsson** [comment]
Nice 💯

**Review by MattiasNilsson — APPROVED** [review]

**Pull request merged**